src/actionStore.ts

```diff
@@ -34,6 +38,7 @@ function updateAction(actionToUpdate: Action) {
     const action = actions.find((a) => a.Id === actionToUpdate.Id)! // we know the element will always be found
 
     action.Label = actionToUpdate.Label
+    action.Elements = actionToUpdate.Elements.map((d) => ({ ...d, Id: d.Id ?? guid() }))
```

Comment:
shouldn't use `d` here. Also, just to clarify, since we will have some elements of the action that already exist, and some might have been added as part of the edit, we either reuse the existing identifier, or create a new one